src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -89,3 +95,21 @@ void Qgs3DMapCanvas::setViewFromTop( const QgsPointXY &center, float distance, f
   float worldY = center.y() - mMap->origin().y();
   mScene->cameraController()->setViewFromTop( worldX, -worldY, distance, rotation );
 }
+
+void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileFormat )
+{
+  if ( !fileName.isEmpty() )
+  {
+    QMetaObject::Connection connection;
+    mCaptureReply = mCapture->requestCapture();
+    connection = connect( mCaptureReply, &Qt3DRender::QRenderCaptureReply::completed, this, [ = ]
+    {
+      disconnect( connection );
```

Comment:
this isn't needed - you're deleting mCaptureReply a few lines down, so the connection will be destroyed then anyway